src/coreclr/jit/emitwasm.cpp

```diff
@@ -186,21 +252,70 @@ size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
 
     switch (insFmt)
     {
+        case IF_NONE:
+            break;
         case IF_OPCODE:
             dst += emitOutputByte(dst, opcode);
             break;
         case IF_BLOCK:
             dst += emitOutputByte(dst, opcode);
-            dst += emitOutputByte(dst, 0x40);
+            dst += emitOutputByte(dst, 0x40 /* block type of void */);
             break;
         case IF_ULEB128:
+        {
             dst += emitOutputByte(dst, opcode);
-            // TODO-WASM: emit uleb128
+            cnsval_ssize_t constant = emitGetInsSC(id);
+            dst += emitOutputULEB128(dst, constant);
             break;
+        }
+        case IF_SLEB128:
+        {
+            dst += emitOutputByte(dst, opcode);
+            cnsval_ssize_t constant = emitGetInsSC(id);
+            dst += emitOutputSLEB128(dst, constant);
+            break;
+        }
+        case IF_F32:
+        {
+            dst += emitOutputByte(dst, opcode);
+            // Reinterpret the bits as a double constant and then truncate it to f32,
+            //  then finally copy the raw truncated f32 bits to the output.
+            cnsval_ssize_t bits = emitGetInsSC(id);
+            double         value;
+            float          truncated;
+            memcpy(&value, &bits, sizeof(double));
+            truncated = (float)value;
```

Comment:
Elsewhere we just use `static_cast` so there's no bit manipulation going on
```suggestion
            float truncated = static_cast<float>(bits);
```

Reply:
I think also that would convert the constant's integral value to a float, which isn't what we want here